ui/app/components/shamir-modal-flow.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * @module ShamirModalFlow
+ * ShamirModalFlow components are used to...
+ *
+ * @example
+ * ```js
+ * <ShamirModalFlow @requiredParam={requiredParam} @optionalParam={optionalParam} @param1={{param1}}/>
+ * ```
+ * @param {object} requiredParam - requiredParam is...
+ * @param {string} [optionalParam] - optionalParam is...
+ * @param {string} [param1=defaultValue] - param1 is...
+ */
+import { inject as service } from '@ember/service';
+import ShamirFlow from './shamir-flow';
+import layout from '../templates/components/shamir-modal-flow';
+
+export default ShamirFlow.extend({
+  layout,
+  store: service(),
+  onClose: () => {},
+  actions: {
+    onCancelClose() {
+      if (this.encoded_token) {
+        this.send('reset');
```

Comment:
nice job figuring this one out. for my own learning, what was the issue? i'm familiar with `this.send()` as a way to send actions from a component to a controller, but from what i understand here `reset` is defined on the `ShamirFlow` component, not on the controller. did you have to use `this.send` here to ensure `ShamirFlow.reset()` was called?

Reply:
There was a `reset` method on the controller (which I had access to with simply `this.reset()` but it wasn't clearing all the necessary values for a full reset, [the _action_ named reset](https://github.com/hashicorp/vault/blob/cbd1369f0f0b65a4607c8bdd9d3ffc068c13a7f1/ui/app/components/shamir-flow.js#L145) did. So I called the action named reset using `this.send`